GSA/digitalgov.gov
# Remove ring header from in-page-nav, fix style bug for h2

## Summary
The “Ring” Hugo shortcode adds it to the “On this page” menu in the right sidebar, but the link doesn’t work.

To replicate: click on the “Case Study” or “Additional Resources” sidebar menu link on:

- [An introduction to digital governance](https://digital.gov/resources/an-introduction-to-digital-governance/)
- [An introduction to inclusion](https://digital.gov/resources/an-introduction-to-inclusion/)
- [An introduction to security and privacy controls](https://digital.gov/resources/an-introduction-to-security-and-privacy-controls/)

Was developed for the HCD Guide.
This only seems to be an issue on Resource pages; not guide or blog pages ([example](https://digital.gov/guides/hcd/design-concepts/design-at-scale/#content-start)).

The ring.html file has an H2 html tag set for the title, but it displays much smaller than an actual H2 header:

`<article class="dg-ring" aria-labelledby="{{ $randomHeaderID }}">
  <h2 id="{{ $randomHeaderID }}" class="dg-ring__title">
    {{- $title -}}
  </h2>
  {{ .Inner | markdownify }}
</article>`

See Trello card 'Fix bug with ring shortcode and menu sidebar'.

### Solution
Filtered out ring class `dg-ring` in `in-page-nagivation.js`. 
`  navItems.forEach((item) => {
    const targetId = item.querySelector("a").getAttribute("href").slice(1);
    const targetElement = document.getElementById(targetId);

    if (targetElement && targetElement.closest(".dg-ring")) {
      const newItem = item;
      newItem.style.display = "none";
    }
  });`

### How To Test

visit any of the preview pages below, check ring headers and style is appropriate
| previews |
|-----------|
| [page 1](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/tc-fix-ring-shortcode/resources/an-introduction-to-digital-governance/) |
| [page 2](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/tc-fix-ring-shortcode/resources/an-introduction-to-inclusion/) |
| [page 3](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/tc-fix-ring-shortcode/resources/an-introduction-to-security-and-privacy-controls/) |

---

### Dev Checklist
